plugins/samples/redirect/plugin.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_redirect]
+#include "absl/strings/str_replace.h"
+#include "proxy_wasm_intrinsics.h"
+
+// This sample may redirects the request for other URL according to the URL
+// requested.
+class MyHttpContext : public Context {
+ public:
+  explicit MyHttpContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t headers,
+                                       bool end_of_stream) override {
+    auto path = getRequestHeader(":path")->toString();
+    // Change the values according to your needs.
+    if (absl::StrReplaceAll({{"/foo/images", "/bar/images"}}, &path) > 0) {
```

Comment:
I don't think this is fully correct, since it would map "/qux/foo/images" to /qux/bar/images", when we only want to rewrite paths under top-level directory "/foo/..." to be "/bar/...".

Suggest instead doing:

```
constexpr std::string_view old_path_prefix = "/foo/";
constexpr std::string_view new_path_prefix = "/bar/";
if (absl::StartsWith(path, old_path_prefix)) {
  std::string new_path = absl::StrCat(new_path_prefix, path.substr(old_path_prefix.length()));
  sendLocalResponse(
    301, "", absl::StrCat("Content moved to ", new_path),
    {{"Location", new_path}});
    return FilterHeadersStatus::ContinueAndEndStream;
}
```

Reply:
Makes sense... Done